src/scripts/main.js

```diff
@@ -1,3 +1,216 @@
+
+function updateRows() {
+  rows = [...tBody.querySelectorAll('tr')];
+}
+
+headers.forEach((header, index) => {
+  header.addEventListener('click', () => {
+    setSortTable(index);
+  });
+});
+
+const setSortTable = (columnIndex) => {
+  updateRows();
+
+  const sortedRows = rows.sort((a, b) => {
+    const aValue = a.cells[columnIndex].textContent.trim();
+    const bValue = b.cells[columnIndex].textContent.trim();
+
+    if (!isNaN(aValue) && !isNaN(aValue)) {
+      return sortOrder[columnIndex] ? aValue - bValue : bValue - aValue;
+    }
+
+    if (columnIndex === 4) {
+      const aSalary = parseFloat(aValue.replace(/[$,]/g, ''));
+      const bSalary = parseFloat(bValue.replace(/[$,]/g, ''));
+
+      return sortOrder[columnIndex] ? aSalary - bSalary : bSalary - aSalary;
+    }
+
+    return sortOrder[columnIndex]
+      ? aValue.localeCompare(bValue)
+      : bValue.localeCompare(aValue);
+  });
+
+  tBody.innerHTML = '';
+  tBody.append(...sortedRows);
+  sortOrder[columnIndex] = !sortOrder[columnIndex];
+};
+
+// Selecting row:
+
+[...tBody.rows].forEach((row) => {
+  row.addEventListener('click', (e) => {
+    const rowsActive = tBody.querySelectorAll('tr.active');
+
+    rowsActive.forEach((rowActive) => {
+      rowActive.classList.remove('active');
+    });
+
+    row.classList.toggle('active');
+  });
+});
+
+// Create form:
+
+const form = document.createElement('form');
+
+form.className = 'new-employee-form';
+
+form.innerHTML = `
+  <label>
+    Name: <input name="name" type="text" data-qa="name" required>
+  </label>
+  <label>
+    Position: <input name="position" type="text" data-qa="position" required>
+  </label>
+  <label>
+    Office:
+    <select name="office" data-qa="office" required>
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>
+    Age: <input name="age" type="number" data-qa="age" required>
+  </label>
+  <label>
+    Salary: <input name="salary" type="number" data-qa="salary" required>
+  </label>
+  <button type="button">Save to table</button>
+`;
+
+body.append(form);
+
+// Add new row:
+
+form.querySelector('button').addEventListener('click', () => {
+  if (!checkFormFields()) {
+    return;
+  }
+
+  const newRow = document.createElement('tr');
+
+  if (
+    form.elements.name.value &&
+    form.elements.position.value &&
+    form.elements.office.value &&
+    form.elements.age.value &&
+    form.elements.salary.value
+  ) {
+    newRow.insertCell(0).textContent = form.elements.name.value.trim();
+    newRow.insertCell(1).textContent = form.elements.position.value.trim();
+    newRow.insertCell(2).textContent = form.elements.office.value.trim();
+    newRow.insertCell(3).textContent = parseInt(form.elements.age.value, 10);
+
+    const salaryFormatted = parseFloat(
```

Comment:
The variable `salaryFormatted` is misspelled as `salaryFormated`. It should be corrected to `salaryFormatted` to improve code readability.